scan.go

```diff
@@ -272,9 +265,53 @@ func scan(pkgPatterns []string, scanDir string) ([]*unit.SourceUnit, error) {
 			Ops:          map[string]*srclib.ToolRef{"depresolve": nil, "graph": nil},
 		})
 	}
-	if err := cmd.Wait(); err != nil {
+
+	return units, nil
+}
+
+func scanForPackages(dir string) ([]*build.Package, error) {
+	var pkgs []*build.Package
+
+	pkg, err := buildContext.ImportDir(dir, 0)
+	if _, isNoGoError := err.(*build.NoGoError); err != nil && !isNoGoError {
 		return nil, err
 	}
+	if err == nil {
+		pkgs = append(pkgs, pkg)
+	}
 
-	return units, nil
+	infos, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	for _, info := range infos {
+		name := info.Name()
+		fullPath := filepath.Join(dir, name)
+		if info.IsDir() && ((name[0] != '.' && name[0] != '_' && name != "testdata") || (strings.HasSuffix(fullPath, "Godeps/_workspace") && !config.SkipGodeps)) {
```

Comment:
Two issues here.

One, you're using string equation with filepaths. This may fail on OSes that don't use forward slash as filepath separator. You can fix it by applying [`filepath.ToSlash`](https://godoc.org/path/filepath#ToSlash).

Two, you're checking for "Godeps/_workspace" suffix but that may incorrectly match something else like "ThisIsFakeGodeps/_workspace". To fix this, you can check for "/Godeps/_workspace" suffix.

Another minor point, will we want to add other special cases for godep-like workspaces (`vendor`, `third_party`, etc.)? If so, we'll want to make this a little more general by factoring out. Not necessary to do right now, just mentioning this thought.